src/data_cleaning.py

```diff
@@ -1272,15 +1276,73 @@ def identify_partial_cems_plants(all_data):
         validate="m:1",
     )
 
+    # Plants with these primary fuel types shouldn't be shaped using CEMS data --
+    # any cems data is likely to be a backup generator and therefore not representative of overall generation.
+    non_cems = ["WND", "WAT", "NUC", "SUN", "GEO"]
+
     # for plants where the data source is eia only, and it exists in partial plant, change the source to partial_cems_plant
+    # but only if the fuel type is consistent with CEMS fuel types
     all_data.loc[
         (all_data["hourly_data_source"] == "eia")
-        & (all_data["partial_plant"] == "both"),
+        & (all_data["partial_plant"] == "both")
+        & (
+            ~all_data["energy_source_code"].isin(non_cems)
+        ),  # CEMS data should only be used to shape if plant is primarilly fossil
         "hourly_data_source",
     ] = "partial_cems_plant"
 
+    # We might have some plants with NaN-ID subplants with mixed hourly methods
+    # due to the hourly source code check above. In those cases, we need to select one methodology -- otherwise we'll double-count the generation
+    # We'll choose the generator with the largest net_generation_mwh and use its methodology.
```

Comment:
Can you provide some summary information about how much generation this method impacts? Does this happen to affect more plants than just the specific situation of backup generators that run more than a handful of hours per year? How common is NA subplants with mixed fuels?

Reply:
This occurs in 74 subplant-months in plants 141, 621, 1943, 2240, 10025, 10823, and 58236. It is not always a case of backup generators; for example, 141 is mostly a natural gas plant with one small solar array with zero generation reported to EIA-923.

Reply:
There is a total of 13034375 MWh over the affected subplant-months